docs/tutorials/04_threads.md

```diff
@@ -0,0 +1,167 @@
+
+The full code for this tutorial is available at: {@link https://github.com/ForstaLabs/forsta-messenger-client/tree/master/examples/threads}
+
+In the last tutorial we configured the client to connect, authenticate, and provision a regular user by way of API calls and the Forsta JWT. In this tutorial we will continue our exploration of the Forsta Messenger Client by introducing some thread operations.
+
+Thread operations are built-in methods that can be used to control every aspect of threads: creation, distribution, attribute designation, and deletion are among the many actions that you will be familiar with by the end of this tutorial.
+
+In order to illustrate just how useful thread operations are, let's imagine a scenario. It's the beginning of October, and the boss has put you in charge of planning a work Halloween party. How do you even start with such a task? Fresh off the previous three tutorials and itching to use your shiny new Forsta client, you decide to create a planning thread that includes the two departments that you plan to invite: sales and support.
+
+Using the distribution tags for sales (@sales) and support (@support), and assuming you also want to add your own department devs (@devs), here's what the code for creating and opening your Halloween Party thread would look like:
+
+```html
+async function onLoaded(client) {
+  const threadId = await client.threadMake("@sales + @devs + @support");
+  await client.threadOpen(threadId); 
+      
+  displayMessenger();
+}
+```
+    
+This is a great start, but you might notice a few things are sub-optimal.
+  1. The `threadMake()` operation doesn't check if a thread matching this distribution already exists. This becomes immediately clear if you run the code again, as a second thread is created.
+  2. There is no title for your thread. This could be confusing for everyone who didn't already know you were going to add them.
+  
+This code fixes these problems by leveraging the `threadEnsure()` and the `threadSetAttribute()` operations. We've also created a default message using `threadSendMessage()` that hopefully explains things a bit better.
+
+```html
+async function onLoaded(client) {
+  const threadId = await client.threadEnsure("@sales + @devs + @support");
+  client.threadSetAttribute(threadId, "title", "Halloween Party");
+  await client.threadOpen(threadId);
+      
+  client.threadSendMessage(threadId,"This is the planning thread for the halloween party! I know it's early, so tomorrow I'll archive this until a week before halloween.");
+      
+  displayMessenger();
+}
+```
+    
+Next, let's focus on actualizing that promise to archive the thread from tomorrow until a week before halloween. This can be done via the `threadArchive()` and the `threadRestore()` commands as well as a few lines of additional javascript. To make things a bit simpler, let's assume that the day that this tutorial takes place on is October 5th (10/5).
```

Comment:
we should be able to link directly to those methods in the API refs.  I don't recall the extra grammar but there is a notation that should allow them to be directly linked.  See the *jsdoc* docs.